src/main/java/com/spring/board/springboard/post/service/PostService.java

```diff
@@ -54,28 +54,43 @@ findPost, new MemberSummaryResponseDto(findPost.getMember())
         );
     }
 
-    public PostDetailResponseDto createPost(PostCreateRequestDto postCreateRequestDto) {
-        Member findMember = memberService.getMember(postCreateRequestDto.memberId());
+    public PostDetailResponseDto createPost(PostCreateRequestDto postCreateRequestDto, String email) {
+        Member findMember = memberService.getMemberByEmail(email);
 
         Post post = postCreateRequestDto.toEntity(findMember);
         postRepository.save(post);
 
         return new PostDetailResponseDto(post, new MemberSummaryResponseDto(findMember));
     }
 
-    public PostDetailResponseDto update(Integer postId, PostCreateRequestDto postCreateRequestDto) {
+    public PostDetailResponseDto update(Integer postId, PostCreateRequestDto postCreateRequestDto, String email) {
         Post findPost = postRepository.findById(postId)
                 .orElseThrow(() -> {
                     throw new NoPostException(
                             MessageFormat.format("(id: {0})에 해당하는 게시물이 없습니다.", postId)
                     );
                 });
 
+        findPost.validateWriter(email);
+
         findPost.change(
                 postCreateRequestDto.title(),
                 postCreateRequestDto.content()
         );
 
         return new PostDetailResponseDto(findPost, new MemberSummaryResponseDto(findPost.getMember()));
     }
+
+    public void deleteOne(Integer postId, String email) {
+        Post findPost = postRepository.findById(postId)
+                .orElseThrow(() -> {
+                    throw new NoPostException(
+                            MessageFormat.format("(id: {0})에 해당하는 게시물이 없습니다.", postId)
+                    );
+                });
+
+        findPost.validateWriter(email);
```

Comment:
쿼리레벨에서 필터링을 해버리면 되지 않을까요?

```suggestion
        Post findPost = postRepository.findByIdAndUserEmail(postId, email)
                .orElseThrow(() -> {
                    throw new NoPostException(
                            MessageFormat.format("(id: {0})에 해당하는 게시물이 없습니다.", postId)
                    );
                });
```